tests/test_fna2d.py

```diff
@@ -727,6 +727,45 @@ def test_against_cutlass_fna(self):
                     is_causal=is_causal,
                 )
 
+    @unittest.expectedFailure
+    @skip_if_cuda_is_not_supported()
+    @skip_if_fna_is_not_supported()
+    def test_against_cutlass_fna(self):
+        problem_sizes = [
+            (1, 1, 3, 3, 16, 3, 3, 1, 1),
+            (1, 1, 8, 10, 16, 3, 5, 1, 2),
+            (1, 2, 15, 20, 32, 5, 15, 3, 1),
+            (1, 2, 17, 19, 32, 9, 7, 1, 1),
+            (1, 2, 17, 19, 32, 9, 7, 1, 2),
+            (1, 2, 16, 16, 15, 9, 7, 1, 1),
+            (1, 2, 16, 16, 34, 9, 7, 1, 2),
+        ]
+        for (
+            B,
+            H,
+            X,
+            Y,
+            D,
+            kernel_size_h,
+            kernel_size_w,
+            dilation_h,
+            dilation_w,
+        ) in problem_sizes:
+            for causal_h, causal_w in product([True, False], [True, False]):
+                kernel_size = (kernel_size_h, kernel_size_w)
+                dilation = (dilation_h, dilation_w)
+                is_causal = (causal_h, causal_w)
+                self._test_all_dtypes(
+                    B=B,
+                    H=H,
+                    X=X,
+                    Y=Y,
+                    D=D,
```

Comment:
Awesome!

Although we probably don't need all of those test cases, the very first one should fail.

If you want to test multiple failing things, they need to be separate tests for all of the failures to get "tested".

Reply:
Okay let me just have a couple tests which test the failing parts.

Reply:
I think I have found a slightly elegant solution for writing tests that are functionally the same, but just differ in problem specs: `subTest()`. Take a look at the subtest API in unittest when you get a moment and let me know if you think it is worthwhile to include in the refactoring effort later: https://docs.python.org/3/library/unittest.html#subtests

Reply:
Yeah let's just have a separate PR if it's going to be a larger refactor of all the unit tests.